### PR TITLE
Add GitHub Action configuration for fontbakery and OpenType sanitizer CI testing

### PR DESCRIPTION
**Pull request by chrissimpkins**
Adds:

- requirements.txt file with pinned fontmake and fontmake specified dependencies, gftools and gftools specified dependencies
- requirements.in file to generate requirements.txt file with the `pip-tools` project `pip-compile` executable
- GitHub Action configuration file for automated remote builds with fontbakery CI testing

### TODO:

- determine an approach to install ttfautohint (apt package manager install vs. build from source with https://github.com/source-foundry/ttfautohint-build vs. ttfautohint-py package?) in Linux environment
- determine approach to web font compiler installation in Linux environment (build from source vs. transition to ttx?)

### CODE REVIEW

**Comment by chrissimpkins** [comment]
You may need to ask upstream repo maintainer to activate GH Actions on downstream repositories so that we can test the Actions workflows here

**Comment by RosaWagner** [comment]
Thanks. I asked upstream to merge my PR so I can merge this one without adding stuff there… but I don't know why they don't do it. Is there something special to do to activate GH actions ?

**Comment by chrissimpkins** [comment]
I believe that this Actions setting allows actions on downstream forks

File: .github/workflows/fontbakery-ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Python build dependency cache lookup
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          # Check for requirements file cache hit
+          key: ${{ runner.os }}-pip-${{ hashFiles('${{ steps.config.outputs.dependpath }}') }}
+      - name: Install Python build dependencies
```
Comment (chrissimpkins) [inline .github/workflows/fontbakery-ci.yml:25]: This step installs your requirements.txt specified Python dependencies

File: .github/workflows/fontbakery-ci.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          update-wheel: "true"
+          update-setuptools: "true"
+      # - name: Build fonts
+      #   run: sources/build.sh
+      # - name: Font Bakery GS profile tests (Expert artifacts)
+      #   uses: f-actions/font-bakery@v1
+      #   with:
+      #     version: "latest"
+      #     subcmd: "check-googlefonts"
```
Comment (chrissimpkins) [inline .github/workflows/fontbakery-ci.yml:36]: This is the location for the fontbakery subcommand.  We need to know what fontbakery profile `gftools qa -f [FONT] -fontbakery` runs

File: .github/workflows/fontbakery-ci.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          path: ~/.cache/pip
+          # Check for requirements file cache hit
+          key: ${{ runner.os }}-pip-${{ hashFiles('${{ steps.config.outputs.dependpath }}') }}
+      - name: Install Python build dependencies
+        uses: py-actions/py-dependency-install@v2
+        with:
+          update-wheel: "true"
+          update-setuptools: "true"
+      # - name: Build fonts
```
Comment (chrissimpkins) [inline .github/workflows/fontbakery-ci.yml:30]: We will need to define how to install ttfautohint before this step of the build

We will need to move webfont compiles to a separate build script and *not* test as part of CI or find a way to compile / install web font compilers used in this build process.

Comment (chrissimpkins) [reply .github/workflows/fontbakery-ci.yml:30]: Options to address these issues documented in the OP

**Comment by chrissimpkins** [comment]
Here is an approach to fontTools based woff and woff2 builds from other build formats:

```python
from fontTools.ttLib import sfnt, TTFont

tt = TTFont("TestFont-Regular.ttf")

# woff
# Note: requires zopfli Python package
tt.flavor = "woff"
sfnt.USE_ZOPFLI = True
tt.save("testfont-regular.woff")

# woff2
# Note: requires brotli Python package
tt.flavor = "woff2"
tt.save("testfont-regular.woff2")
```

If https://github.com/fonttools/fonttools/pull/2066 is accepted in fontTools you will be able to toggle zopfli compression on your woff files through a TTFont.save() parameter.  

This should allow you to move away from the compiled compiler dependencies.  This should make remote build workflows more straightforward.

**Comment by RosaWagner** [comment]
I had the confirmation from Marc that gftools qa uses check-googlefonts profile for Fontbakery :)

**Comment by RosaWagner** [comment]
I think your last solution for the web fonts is indeed more practical than creating a separate build script as was doing for other projects

**Pull request merged**